sdk/core/azure-core-amqp/docs/architecture.md

```diff
@@ -0,0 +1,53 @@
+# Architecture Docs
+
+## Qpid Proton-J Integration
+
+Qpid Proton-J publishes events and messages via its event-driven process called [`Reactor`][Reactor]. `azure-core-amqp`
+hooks into Qpid Proton-J Reactor via `Handlers`.  [`BaseHandler`][BaseHandler] contains all of the events that can be
+listened to.  `Handlers` can be associated with classes implementing the `Extendable` interface via
+`BaseHandler.setHandler(Extendable, Handler)`.  Proton-J `Connection`, `Session`, `Link`, `Sender` and `Receiver` are
+all `Extendable`.
+
+The UML diagram below shows this relationship; the interfaces shown in green are Proton-J classes. Each Proton-J
+instance (ie. `Connection`, `Session`, `Sender`, `Receiver`) is associated with one corresponding `*Handler`.  Each
```

Comment:
will the `Reactor` object also own the `Connection` object whichh can be specicifed when registering user-handler to handle events from that connection object? I mean my main question is - who is responsible to own and manage the connection? user or reactor.